docs/working-with-mira/README.md

```diff
@@ -0,0 +1,162 @@
+# Working With Mira
+## Mira CLI
+  - Developer friendly CLI, simply run `npx mira`.
+      - init - option to initialize mandatory files.
+      - deploy - option to deploy your application directly from your computer.
+      - cicd - option to deploy shared CI Pipeline
+      - docs - option to run a docsify webserver and see the Mira docs.
+
+## Deploying S3 Bucket Content Only
+(*Experimental Feature*)
+
+Sometimes you want to only deploy changes made to S3 content, such as with frontends you wish to deploy.  It can be cumbersome to go through a stack changeset as this can take minutes.  You can use the `--s3-only` flag to attempt to deploy any assets that CDK would otherwise push up to S3 buckets.  This can shrink deployment times down to 15-30 seconds and allows you to share S3-connected assets easily.  To do this, you might run a command like:
+
+`npx mira deploy myStackName --file=path/to/stackFile.js --s3-only`
+
+Note, that this command will only work if you have already deployed this stack once before!  If it is successful, you should see the console log the HTTP-accessible location of your files:
+
+![quick-deploy]
+
+## Developing with Mira locally
+
+You may want to author changes to the Mira package and immediately use those
+changes in your Mira application.  You can do this using the following syntax
+from your local Mira directory:
+
+```
+node scripts/link.js [APPLICATION_PATH]
+```
+
+For instance, if your application is at `../app` you'd run the script:
+
+```
+node scripts/link.js ../app
+```
+
+If you have an existing installation or Mira dependency installed in `../app` 
+then Mira will copy it to `../app/mira.old`.  To reinstall a remote version of
+Mira from GitHub or NPM just delete your `node_modules` folder and re-run 
+`npm install` in your application directory.
```

Comment:
Could/should add below this paragraph:

Each time you make local changes to Mira, remember you will need to run `npm run build` within the Mira package.  Likewise, in your application: if you change your package.json, add or remove a dependency you will impact Mira and will need to re-run `npm install` and `node scripts/link [APPLICATION_PATH`.  This is because Mira symlinks itself into the application `node_modules` folder and running an `npm install` in the application will wipe `node_modules` it thinks are not relevant, i.e. it will delete Mira's dependencies from its `node_modules`.  This caveat may be later automated but is important to note.

Reply:
added